### DISCUSSION TITLE
How to use system certificates?

### DISCUSSION THREAD

**Discussion by pikondrej** [Ericsson/codechecker#3852] [Q&A]
I have the server side running and use basically self-signed certificate. I have the certificate properly installed in the system. Could you please tell me what can I do to make CodeChecker use the system certificates / or / use specific certificate / or / disable SSL verification?
Thank you

EDIT:
I wasn't clear. The server side is running fine. I am having trouble connecting from another machine using the CodeChecker as client. The login cmd ends up with:
`certificate verify failed: unable to get local issuer certificate`
The cert should be properly installed on the client side system (docker Ubuntu 20.04).
The connection works when specifying the system certificates or via wget/curl:

```
import http.client;
import ssl;

context = ssl.SSLContext(ssl.PROTOCOL_TLS);
context.load_verify_locations(cafile='/etc/ssl/certs/ca-certificates.crt', capath=None, cadata=None);
connection = http.client.HTTPSConnection('codechecker.x.com:443', context=context);

connection.request('GET', '/');
response = connection.getresponse();
print(response.status);
```
I tried all I could find to disable ssl check or set up correct path. `certifi.where()` returns the `/etc/ssl/certs/ca-certificates.crt`. I tried exporting `CURL_CA_BUNDLE`, `REQUESTS_CA_BUNDLE` and `SSL_CERT_FILE`, but nothing works. I am unfamiliar with python, so I don't know what is its behavior when it comes to certificates and I can't recognize if CodeChecker is doing something "different" perhaps.

**Comment by whisperity** [comment]
The easy answer would be to either copy or symlink the `cert.pem` and `key.pem` into the server's *config-directory*, where CodeChecker searches for it.
Symbolic links would allow the system-wide certificate to be updated without having to do the copy again.

**Reply by pikondrej** [reply]
Sorry, I wasn't clear. The server side is running fine. I am having trouble connecting from another machine using the CodeChecker as client. The login cmd ends up with:
`certificate verify failed: unable to get local issuer certificate`
The cert should be properly installed on the client side system (docker Ubuntu 20.04).
The connection works when specifying the system certificates:

```
import http.client;
import ssl;

context = ssl.SSLContext(ssl.PROTOCOL_TLS);
context.load_verify_locations(cafile='/etc/ssl/certs/ca-certificates.crt', capath=None, cadata=None);
connection = http.client.HTTPSConnection('codechecker.x.com:443', context=context);

connection.request('GET', '/');
response = connection.getresponse();
print(response.status);
```
I tried all I could find to disable ssl check or set up correct path. `certifi.where()` returns the `/etc/ssl/certs/ca-certificates.crt`. I tried exporting `CURL_CA_BUNDLE`, `REQUESTS_CA_BUNDLE` and `SSL_CERT_FILE`, but nothing works. I am unfamiliar with python, so I don't know what is its behaviour when it comes to certificates and I can't recognize if CodeChecker is doing something "different" perhaps.
